www/src/components/publisher/CreatePublisherModal.tsx

```diff
@@ -0,0 +1,281 @@
+import { useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Button, Flex, Modal } from 'honorable'
+import {
+  Alert,
+  FormField,
+  Input,
+  ModalActions,
+  ModalHeader,
+} from 'pluralsh-design-system'
+import { useMutation } from '@apollo/client'
+import { useFilePicker } from 'react-sage'
+import isArray from 'lodash/isArray'
+
+import { generatePreview } from '../../utils/file'
+
+import IconUploadPreview from '../utils/IconUploadPreview'
+
+import { CREATE_PUBLISHER_MUTATION } from './queries'
+
+type CreatePublisherModalProps = {
+  open: boolean
+  onClose: () => void
+}
+
+type IconUploadType = {
+  file: File | null;
+  previewUrl: string | null;
+}
+
+function CreatePublisherModal({ open, onClose }: CreatePublisherModalProps) {
+  const [iconUpload, setIconUpload] = useState<IconUploadType>({ file: null, previewUrl: null })
+  const [name, setName] = useState('')
+  const [description, setDescription] = useState('')
+  const [website, setWebsite] = useState('')
+  const [documentation, setDocumentation] = useState('')
+  const [github, setGithub] = useState('')
+  const [discord, setDiscord] = useState('')
+  const [slack, setSlack] = useState('')
+  const [twitter, setTwitter] = useState('')
+  const [preMutationError, setPreMutationError] = useState('')
+  const [errors, setErrors] = useState({
+    website: false,
+    documentation: false,
+    github: false,
+    discord: false,
+    slack: false,
+    twitter: false,
+  })
+  const navigate = useNavigate()
+
+  const iconPicker = useFilePicker({
+    minImageWidth: 64,
+    maxImageWidth: 512,
+    minImageHeight: 64,
+    maxImageHeight: 512,
+  })
+  const iconPickerInputOpts = {
+    multiple: false,
+    accept: 'image/jpeg,image/png',
+  }
+
+  const [mutation, { data, loading, error }] = useMutation(CREATE_PUBLISHER_MUTATION, {
+    variables: {
+      attributes: {
+        name,
+        description,
+        ...(iconUpload.file ? { avatar: iconUpload.file } : {}),
+      },
+    },
+  })
+
+  useEffect(() => {
+    const file = isArray(iconPicker?.files) && iconPicker?.files[0]
+
+    if (!file) return
+
+    const reader = generatePreview(file, (file: IconUploadType) => {
+      setIconUpload(file)
+    })
+
+    return () => {
+      reader.abort()
+    }
+  }, [iconPicker.files])
+
+  useEffect(() => {
```

Comment:
this should be done in an `onCompleted` callback for the useMutation hook

Reply:
On it.